darts/utils/data/training_dataset.py

```diff
@@ -171,6 +167,41 @@ def _memory_indexer(self,
 
         return past_start, past_end, future_start, future_end, cov_start, cov_end
 
+    def _generate_covariates(self,
```

Comment:
It might be a bit clearer to renamed into sth like `_add_encodings_to_covariates()` ?

Reply:
True! +1